animations/hero-animations/index.md

```diff
@@ -0,0 +1,670 @@
+  property.
+
+### Photo class
+
+The Photo class builds the widget tree that holds the image:
+
+<!-- skip -->
+{% prettify dart %}
+class Photo extends StatelessWidget {
+  Photo({ Key key, this.photo, this.color, this.onTap }) : super(key: key);
+
+  final String photo;
+  final Color color;
+  final VoidCallback onTap;
+
+  Widget build(BuildContext context) {
+    return [[highlight]]new Material([[/highlight]]
+      // Slightly opaque color appears where the image has transparency.
+      [[highlight]]color: Theme.of(context).primaryColor.withOpacity(0.25),[[/highlight]]
+      child: [[highlight]]new InkWell([[/highlight]]
+        onTap: [[highlight]]onTap,[[/highlight]]
+        child: [[highlight]]new Image.asset([[/highlight]]
+            photo,
+            fit: BoxFit.contain,
+          )
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+Key information:
+
+* The Inkwell captures the tap gesture.
+  The calling function passes the `onTap()` function to the
+  Photo's constructor.
+
+* During flight, the InkWell draws its splash on its first Material
+  ancestor.
+
+* The Material widget has a slightly opaque color, so the
+  transparent portions of the image are rendered with color.
+  This ensures that the circle-to-square transition is easy to see,
+  even for images with transparency.
+
+* The Photo class does not include the Hero in its widget tree.
+  For the animation to work, the hero
+  wraps the RadialExpansion widget, which wraps the hero.
+
+### RadialExpansion class
+
+The RadialExpansion widget, the core of the demo, builds the
+widget tree that clips the image during the transition.
+The clipped shape results from the intersection of a circular clip
+(that grows during flight),
+with a rectangular clip (that remains a constant size throughout).
+
+To do this, it builds the following widget tree:
+
+<img src="images/radial-expansion-class.png" alt="widget tree for the RadialExpansion widget">
+
+Here's the code:
+
+<!-- skip -->
+{% prettify dart %}
+class RadialExpansion extends StatelessWidget {
+  RadialExpansion({
+    Key key,
+    this.maxRadius,
+    this.child,
+  }) : [[highlight]]clipRectSize = 2.0 * (maxRadius / math.SQRT2),[[/highlight]]
+       super(key: key);
+
+  final double maxRadius;
+  final clipRectSize;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {[[/highlight]]
+    return [[highlight]]new ClipOval([[/highlight]]
+      child: [[highlight]]new Center([[/highlight]]
+        child: [[highlight]]new SizedBox([[/highlight]]
+          width: clipRectSize,
+          height: clipRectSize,
+          child: [[highlight]]new ClipRect([[/highlight]]
+            child: [[highlight]]child,[[/highlight]]  // Photo
+          ),
+        ),
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+Key information:
+
+<ul markdown="1">
+<li markdown="1">The hero wraps the RadialExpansion widget.
+</li>
+<li markdown="1">As the hero flies, the RadialExpansion widget's
```

Comment:
I think it would be helpful to explain that as the hero flies, its size changes, and because it constrains its child's size, the RadialExpansion widget changes size to match.